memo.rb

```diff
@@ -2,47 +2,57 @@
 # frozen_string_literal: true
 
 require 'csv'
+require 'pg'
 require 'securerandom'
 require 'sinatra'
 require 'sinatra/reloader'
 
-HEADERS = %w[
-  id
-  title
-  text
-].freeze
-
-CSV_PATH = 'memos.csv'
+DB_NAME = 'memoapp'
 
 helpers do
   def h(text)
     Rack::Utils.escape_html(text)
   end
 end
 
-def get_memo(id)
-  read_memos.find { |memo| memo['id'] == id }
+def connect_db
+  PG.connect(dbname: DB_NAME)
 end
 
-def read_memos
-  CSV.read(CSV_PATH, headers: true).map(&:to_hash)
+def select_one(id)
+  conn = connect_db
+  conn.exec_params('SELECT * FROM memo WHERE id = $1', [id]).first
 end
 
-def write_memos(edited_memos)
-  CSV.open(CSV_PATH, 'w', headers: HEADERS, write_headers: true) do |memos|
-    edited_memos.each { |edited_memo| memos << edited_memo }
-  end
+def select_all
+  conn = connect_db
+  conn.exec_params('SELECT * FROM memo ORDER BY created_at')
+end
+
+def insert_memo(memo)
+  conn = connect_db
+  conn.exec_params('INSERT INTO memo (title, main_text) VALUES ($1, $2) RETURNING id', memo.values_at(:title, :main_text))
```

Comment:
```suggestion
  conn.exec_params(
    'INSERT INTO memo (title, main_text) VALUES ($1, $2) RETURNING id', 
    memo.values_at(:title, :main_text)
  )
```

ちょっと横に長いので改行した方が読みやすいかも